lib/hammer_cli_katello/host_collection.rb

```diff
@@ -51,15 +51,40 @@ class InfoCommand < HammerCLIKatello::InfoCommand
     end
 
     class HostsCommand < HammerCLIKatello::ListCommand
-      resource :host_collections, :hosts
+      resource :hosts, :index
       command_name "hosts"
 
+      option "--id", "HOST_COLLECTION_ID", _("Host Collection ID"), :attribute_name => :option_host_collection_id
+      option "--name", "HOST_COLLECTION_NAME", _("Host Collection Name"), :attribute_name => :option_host_collection_name
+
+      validate_options do
+        host_collection_options = [:option_host_collection_id, :option_host_collection_name]
+        any(*host_collection_options).required
+      end
+
+      def host_collection_options
+        {
+          "option_name" => option_host_collection_name,
+          "option_organization_name" => option_organization_name,
+          "option_organization_id" => option_organization_id,
+          "option_organization_label" => option_organization_label
+        }
+      end
+
+      def request_params
+        params = super
+        collection_id = resolver.host_collection_id(host_collection_options) if option_host_collection_name
+        collection_id = option_host_collection_id if option_host_collection_id
+        params['search'] = "host_collection_id=#{collection_id}"
+        params
+      end
+
```

Comment:
I think this works, but the test API data doesn't seem to be properly mirroring the actual API. You don't need to limit the options passed to `host_collection_id()`:

```ruby
   class HostsCommand < HammerCLIKatello::ListCommand
      resource :hosts, :index
      command_name "hosts"

      option "--id", "HOST_COLLECTION_ID", _("Host Collection ID"),
        attribute_name: :option_host_collection_id
      option "--name", "HOST_COLLECTION_NAME", _("Host Collection Name"),
        attribute_name: :option_host_collection_name

      def request_params
        params = super
        option_host_collection_id ||= resolver.host_collection_id(options.merge(resolver
          .scoped_options("host_collection", options))) if option_host_collection_name
        params['search'] = "host_collection_id=#{option_host_collection_id}"
        params
      end

      output do
        field :id, _("ID")
        field :name, _("Name")
      end

      build_options { |o| o.expand(:all).including(:organizations) }
    end
```